### PR TITLE
Add GitLab CI pipeline.

### PR DESCRIPTION
**Pull request by olupton**
This introduces a GitLab CI pipeline that runs on BBP compute resources. 

This initial version:
- builds CoreNEURON.
- runs the CoreNEURON CTest suite.
- builds NEURON, using the build of CoreNEURON as an external dependency.
- runs the NEURON CTest suite.

Where everything is build/run on CPU using the Intel compiler.

To make this work, some fixes were needed:
- more variables describing the CoreNEURON build configuration are now exported to the installed `coreneuron-config.cmake` file.
- the number of processors required by some CoreNEURON tests is now declared.

CI_BRANCHES:NEURON_BRANCH=master,

### CODE REVIEW

File: .gitlab-ci.yml
Comment (pramodk) [inline .gitlab-ci.yml:38]: We have use this approach already for Jenkins so same can be used?

**Review by pramodk — COMMENTED** [review]
Apart from todos, this looks good to me.

File: .gitlab-ci.yml
Comment (olupton) [reply .gitlab-ci.yml]: I think we can re-use some commands from Jenkins, but it will also need some new plumbing.

This can be addressed later. In the short term we can just modify `.gitlab-ci.yml` in open PRs that are waiting for changes to be merged into NEURON, as was done here.

Comment (olupton) [reply .gitlab-ci.yml]: I opened #505 to track this.

File: .gitlab-ci.yml
Comment (pramodk) [inline .gitlab-ci.yml:3]: q : when some some changes these common template file, are dependent CI projects are triggered so that there is no regression?

File: .gitlab-ci.yml
Comment (pramodk) [inline .gitlab-ci.yml:14]: just curious : this would be the case for all GitHub projects I assume? if so, could this be added somewhere else and inherited from there?

**Review by pramodk — APPROVED** [review]
LGTM

File: .gitlab-ci.yml
Comment (olupton) [reply .gitlab-ci.yml:3]: Good question. It doesn't happen automatically, but @matz-e already set up the CI of `hpc/gitlab-pipelines` to trigger the CI of `hpc/touchdetector` as a regression test. Once this PR is merged I think we should add CoreNEURON's CI alongside TouchDetector's to improve the test coverage.

File: .gitlab-ci.yml
Comment (olupton) [reply .gitlab-ci.yml]: Good suggestion, done in c1fd1dd9fd41e4588088265bf0da9e41578f34e2 -- unfortunately this introduces another TODO; wait for an external MR to be merged.

**Review by iomaganaris — APPROVED** [review]

**Pull request merged**
